src/useCapabilitiesSafe.ts

```diff
@@ -0,0 +1,36 @@
+import type { Chain } from 'viem';
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+import type { WalletCapabilities } from './types';
+
+export function useCapabilitiesSafe({
```

Comment:
why not just `useCapabilities`? like we did for other similar cases.

Reply:
need to add handling for wallets that do not support `useCapabilities`

Reply:
also used as a helper to automatically parse out the capabilities `chainId` we are currently on